test/mechanics/constraintsPde/ConstraintCompanionPde.cpp

```diff
@@ -0,0 +1,195 @@
+#include "BoostUnitTest.h"
+#include <fakeit.hpp>
+//#include <cmath>
+//#include "mechanics/groups/Group.h"
+#include "mechanics/constraintsPde/ConstraintCompanion.h"
+
+using namespace fakeit;
+using namespace NuTo;
+using namespace NuTo::ConstraintPde;
+
+
+//// Helper functions for comparisons
+namespace NuTo
+{
+namespace ConstraintPde
+{
+bool operator==(const Term& a, const Term& b)
+{
+    if (&(a.GetNode()) != &(b.GetNode()))
+        return false;
+    if (a.GetCoefficient() != b.GetCoefficient())
+        return false;
+    if (a.GetComponent() != b.GetComponent())
+        return false;
+    return true;
+}
+
+bool operator==(const Equation& a, const Equation& b)
+{
+    if (a.GetRhs(0.0) != b.GetRhs(0.0))
+        return false;
+    if (a.GetRhs(42.0) != b.GetRhs(42.0))
+        return false;
+    if (a.GetTerms() != b.GetTerms())
```

Comment:
This checks the order of terms, right? Do we want that? The first term somehow determines the dependent dof and is important. The order of the other ones is IMO not important.

Reply:
The order is not important, but this is just a helper function for the test (I just took it from the original testfile). I mean, if somebody changes the internals of the companion so that the current order is messed up, the test will probably fail, even though might not. Other option would be to compare every term of A to every other term of B etc.